pkg/apis/servicecatalog/v1beta1/conversion_test.go

```diff
@@ -187,3 +192,129 @@ func runTestCases(t *testing.T, cases []testcase, testFuncName string, testFunc
 		}
 	}
 }
+
+func TestConvert_v1beta1_CatalogRestrictions_To_servicecatalog_CatalogRestrictions_AndBack(t *testing.T) {
+	originalIn := CatalogRestrictions{
+		ServiceClass: []string{"name not in (foo)"},
+		ServicePlan:  []string{"name == bar", "externalName==baz"},
+	}
+	var originalOut servicecatalog.CatalogRestrictions
+
+	Convert_v1beta1_CatalogRestrictions_To_servicecatalog_CatalogRestrictions(&originalIn, &originalOut, nil)
+
+	var convertedOut CatalogRestrictions
+
+	Convert_servicecatalog_CatalogRestrictions_To_v1beta1_CatalogRestrictions(&originalOut, &convertedOut, nil)
+
+	// original in and converted out should match, but string formatting and order modifications are  allowed.
+
+	for _, r := range convertedOut.ServiceClass {
+		if !findInRequirementsIgnoreSpaces(r, originalIn.ServiceClass) {
+			t.Fail()
+		}
+	}
+
+	for _, r := range convertedOut.ServicePlan {
+		if !findInRequirementsIgnoreSpaces(r, originalIn.ServicePlan) {
+			t.Fail()
+		}
+	}
+}
+
+func findInRequirementsIgnoreSpaces(requirement string, requirements []string) bool {
+	find := strings.Replace(requirement, " ", "", -1)
+	for _, r := range requirements {
+		found := strings.Replace(r, " ", "", -1)
+		if find == found {
+			return true
+		}
+	}
+	return false
+}
+
+func TestConvertClusterServiceClassToProperties(t *testing.T) {
+	cases := []struct {
+		name string
+		sc   *ClusterServiceClass
+		json string
+	}{
+		{
+			name: "nil object",
+			json: "{}",
+		},
+		{
+			name: "normal object",
+			sc: &ClusterServiceClass{
+				ObjectMeta: metav1.ObjectMeta{Name: "service-class"},
+				Spec: ClusterServiceClassSpec{
+					CommonServiceClassSpec: CommonServiceClassSpec{
+						ExternalName: "external-class-name",
+						ExternalID:   "external-id",
+					},
+				},
+			},
+			json: "{\"name\":\"service-class\",\"spec.externalID\":\"external-id\",\"spec.externalName\":\"external-class-name\"}",
```

Comment:
You can use a backticks string here to make this a lot more readable, just fyi

Reply:
nice! thanks, I had not tried yet